subprojects/core/build.gradle.kts

```diff
@@ -204,6 +205,11 @@ tasks.test {
     setForkEvery(200)
 }
 
+// Disable annotation processing for Groovy, as it is not compatible with Groovy IC
+tasks.withType<GroovyCompile>().configureEach {
+    options.compilerArgs.add("-proc:none")
```

Comment:
This line makes the Groovy compiler ignore the annotation processor, so technically it is used on Java sources only.
By the way, the missing support is tracked in https://github.com/gradle/gradle/issues/9871
Do you have another way to set this up in mind?

Reply:
> I would like to get rid of `options.compilerArgs.add("-proc:none")` for GroovyCompile if possible, but if it's not possible then okay.

I think it might be possible if we leave only Groovy test sources in the subproject, as the issue I linked to mentions joint Java+Groovy compilation. 
Disabling Groovy IC does not seem to be a good option to me, especially since the `:core` test sources are massive.